tests/integration/Infrastructure/DashboardBundle/Validator/Constraints/ValidClientIdValidatorTest.php

```diff
@@ -0,0 +1,167 @@
+        // record in the service registry with the given entityId.
+        $this->mockHandler->append(new Response(200, [], '[]'));
+
+        $entityCommand = m::mock(SaveOidcEntityCommand::class);
+        $entityCommand->shouldReceive('getMetadataUrl')->andReturn('https://www.domain.org');
+        $entityCommand->shouldReceive('isForProduction')->andReturn(false);
+        $entityCommand->shouldReceive('getId')->andReturn(1);
+
+        $this->setRoot($entityCommand);
+
+        $this->repository->shouldReceive('findById')
+            ->andReturn(null);
+
+        $this->validator->validate('https://sub.domain.org', new ValidClientId());
+
+        $this->assertNoViolation();
+    }
+
+    public function test_success_for_production()
+    {
+        // An empty array response is Manage's way of telling you there where no results, in this case there is no
+        // record in the service registry with the given entityId.
+        $this->mockHandler->append(new Response(200, [], '[]'));
+
+        $entityCommand = m::mock(SaveOidcEntityCommand::class);
+        $entityCommand->shouldReceive('getMetadataUrl')->andReturn('https://www.domain.org');
+        $entityCommand->shouldReceive('isForProduction')->andReturn(true);
+        $entityCommand->shouldReceive('getId')->andReturn(1);
+
+        $this->setRoot($entityCommand);
+
+        $this->repository->shouldReceive('findById')
+            ->andReturn(null);
+
+        $this->validator->validate('https://sub.domain.org', new ValidClientId());
+
+        $this->assertNoViolation();
+    }
+
+    public function test_empty_client_id()
+    {
+        $entityCommand = m::mock(SaveOidcEntityCommand::class);
+
+        $this->setRoot($entityCommand);
+
+        $this->validator->validate(null, new ValidClientId());
+
+        $violations = $this->context->getViolations();
+        $this->assertCount(1, $violations);
+        $this->assertEquals(
+            'validator.entity_id.invalid_url',
+            $violations->get(0)->getMessageTemplate(),
+            'Expected certain violation but dit not receive it.'
+        );
+    }
+
+    public function test_invalid_domain()
+    {
+        $domainA = 'invalid domain\.com';
+        $domainB = 'domain.org';
+
+        $entityCommand = m::mock(SaveOidcEntityCommand::class);
+        $entityCommand->shouldReceive('isForProduction')->andReturn(false);
+
+        $this->setRoot($entityCommand);
+
+        $constraint = new ValidEntityId();
+        $this->validator->validate('https://sub.' . $domainB, $constraint);
+
+        $violations = $this->context->getViolations();
+        $this->assertCount(1, $violations);
+        $this->assertEquals(
+            'validator.entity_id.invalid_url',
+            $violations->get(0)->getMessageTemplate(),
+            'Expected certain violation but dit not receive it.'
+        );
+    }
+
+    public function test_invalid_client_id_url()
```

Comment:
It might be a nice addition to add a dataProvider that enables us to test more invalid URL's than just one. But has low priority in my opinion.

Reply:
For now I think it will be sufficient but if there should be a change in the feature I will keep this in mind.